syntax = "proto3";

package scalogreplication;

// Service definition for Scalog replication
service ScalogReplicationService {
  rpc Replicate (ScalogReplicationRequest) returns (ScalogReplicationResponse) {}
}

// Request message containing the data to be replicated
message ScalogReplicationRequest {
  int64 offset = 1;
  int64 size = 2;
  int64 num_msg = 3;
  bytes data = 4;
}

// Response message with the result of the replication operation
message ScalogReplicationResponse {
  bool success = 1;
}